### PR TITLE
T5700: Use HttpRequestFactory

### PR DESCRIPTION
**Pull request by Universal-Omega**

### CODE REVIEW

File: includes/RottenLinks.php [hunk 1]
```diff
@@ -4,17 +4,19 @@
...
+		$request = $services->getHttpRequestFactory()->create(
+			$url, [ 
+				'method' => 'HEAD', // return headers only
+				'timeout' => $config->get( 'RottenLinksCurlTimeout' ),
+				'userAgent' => 'RottenLinks, MediaWiki extension (https://github.com/miraheze/RottenLinks), running on ' . $config->get( 'Server' )
+			],
+			__METHOD__
+		)->execute();
+
+		return (int)$request->getStatusValue()->getValue();
```
Comment (Southparkfan) [inline includes/RottenLinks.php:20]: This or https://phabricator.miraheze.org/T5700#145078 will both work, but this avoids adding another variable ($reqexec). +1.

Comment (Universal-Omega) [reply includes/RottenLinks.php:20]: Thanks! Just to clarify, you'd rather me keep my current method then the other one to avoid the extra variable?

Comment (Southparkfan) [reply includes/RottenLinks.php:20]: Your method is fine.

Comment (Universal-Omega) [reply includes/RottenLinks.php:20]: Thanks!

**Pull request merged**